### ISSUE TITLE
Analysis does not find a Typestate Error

### ISSUE THREAD

**Issue by johspaeth** [CROSSINGTUD/CryptoAnalysis#141]
On the following example, the analysis does not detect the typestate error as of the line that is commented in.

```
public class SecUtils {
	private Cipher cipher;

	public SecUtils() throws NoSuchAlgorithmException, NoSuchPaddingException {
		this.cipher = Cipher.getInstance("AES/ECB/PKCS5Padding");
	}
	
	public byte[] encrypt(String data) throws IllegalBlockSizeException, BadPaddingException, GeneralSecurityException {
		KeyGenerator keyGen = KeyGenerator.getInstance("AES");
//		keyGen.init(128);
		SecretKey key = keyGen.generateKey();
		cipher.init(Cipher.ENCRYPT_MODE, key);
		return cipher.doFinal(data.getBytes());
	}
}

```

**Comment by johspaeth** [comment]
Incorrectly opened issue. Is according to [ORDER clause of rule](https://github.com/CROSSINGTUD/Crypto-API-Rules/blob/015a6fc2cbbf36b14bfc96368cf1b21617c578fc/JavaCryptographicArchitecture/src/KeyGenerator.cryptsl#L24).
